hwy/ops/generic_ops-inl.h

```diff
@@ -3117,6 +3117,321 @@ HWY_API V ReverseBits(V v) {
+                                        idx_2_tag, idx_3_tag);
+  return Per4LaneBlockShufTblLookup(lane_size_tag, v, idx);
+}
+
+template <size_t kIdx0, size_t kIdx1, size_t kIdx2, size_t kIdx3, class V,
+          HWY_IF_LANES_LE_D(DFromV<V>, 2)>
+HWY_INLINE V Per4LaneBlockShuffle(hwy::SizeTag<kIdx0> idx_0_tag,
+                                  hwy::SizeTag<kIdx1> idx_1_tag,
+                                  hwy::SizeTag<kIdx2> /*idx_2_tag*/,
+                                  hwy::SizeTag<kIdx3> /*idx_3_tag*/, V v) {
+  return Per2LaneBlockShuffle(idx_0_tag, idx_1_tag, v);
+}
+
+template <size_t kIdx0, size_t kIdx1, class V, HWY_IF_LANES_GT_D(DFromV<V>, 2)>
+HWY_INLINE V Per4LaneBlockShuffle(hwy::SizeTag<kIdx0> idx_0_tag,
+                                  hwy::SizeTag<kIdx1> idx_1_tag,
+                                  hwy::SizeTag<kIdx0 + 2> /*idx_2_tag*/,
+                                  hwy::SizeTag<kIdx1 + 2> /*idx_3_tag*/, V v) {
+  return Per2LaneBlockShuffle(idx_0_tag, idx_1_tag, v);
+}
+
+#if HWY_HAVE_FLOAT64
+template <class V>
+HWY_INLINE VFromD<RepartitionToWide<DFromV<V>>> Per4LaneBlockShufCastToWide(
+    hwy::FloatTag /* type_tag */, hwy::SizeTag<4> /* lane_size_tag */, V v) {
+  const DFromV<decltype(v)> d;
+  const RepartitionToWide<decltype(d)> dw;
+  return BitCast(dw, v);
+}
+#endif
+
+template <size_t kLaneSize, class V>
+HWY_INLINE VFromD<RepartitionToWide<RebindToUnsigned<DFromV<V>>>>
+Per4LaneBlockShufCastToWide(hwy::FloatTag /* type_tag */,
+                            hwy::SizeTag<kLaneSize> /* lane_size_tag */, V v) {
+  const DFromV<decltype(v)> d;
+  const RebindToUnsigned<decltype(d)> du;
+  const RepartitionToWide<decltype(du)> dw;
+  return BitCast(dw, v);
+}
+
+template <size_t kLaneSize, class V>
+HWY_INLINE VFromD<RepartitionToWide<DFromV<V>>> Per4LaneBlockShufCastToWide(
+    hwy::NonFloatTag /* type_tag */,
+    hwy::SizeTag<kLaneSize> /* lane_size_tag */, V v) {
+  const DFromV<decltype(v)> d;
+  const RepartitionToWide<decltype(d)> dw;
+  return BitCast(dw, v);
+}
+
+template <class V, HWY_IF_LANES_GT_D(DFromV<V>, 2),
+          HWY_IF_T_SIZE_ONE_OF_V(V, (1 << 1) | (1 << 2) |
+                                        (HWY_HAVE_INTEGER64 ? (1 << 4) : 0))>
+HWY_INLINE V Per4LaneBlockShuffle(hwy::SizeTag<0> /*idx_0_tag*/,
+                                  hwy::SizeTag<1> /*idx_1_tag*/,
+                                  hwy::SizeTag<0> /*idx_2_tag*/,
+                                  hwy::SizeTag<1> /*idx_3_tag*/, V v) {
+  const DFromV<decltype(v)> d;
+  const auto vw = Per4LaneBlockShufCastToWide(
+      hwy::IsFloatTag<TFromV<V>>(), hwy::SizeTag<sizeof(TFromV<V>)>(), v);
+  return BitCast(d, DupEven(vw));
+}
+
+template <class V, HWY_IF_LANES_GT_D(DFromV<V>, 2),
+          HWY_IF_T_SIZE_ONE_OF_V(V, (1 << 1) | (1 << 2) |
+                                        (HWY_HAVE_INTEGER64 ? (1 << 4) : 0))>
+HWY_INLINE V Per4LaneBlockShuffle(hwy::SizeTag<2> /*idx_0_tag*/,
```

Comment:
And here also: if idx3 is the most-significant, wouldn't Reverse2 idx3..0 = 2301? (idx3=2)